File: 03_hash/sha256/base64url/main.go
package main

import (
	"crypto/sha256"
	"encoding/base64"
	"fmt"
)

func main() {
	text := "Hello world, mac was here"
	hasher := sha256.New()
	hasher.Write([]byte(text))
	hash := base64.RawURLEncoding.EncodeToString(hasher.Sum(nil))
	fmt.Println(hash)
}

File: 03_hash/sha256/base64url/main.cs
var text = "Hello world, mac was here";

var hash = Convert.ToBase64String(System.Security.Cryptography.SHA256.Create().ComputeHash(System.Text.Encoding.UTF8.GetBytes(text))).TrimEnd('=').Replace('+', '-').Replace('/', '_');

Console.WriteLine(hash);
